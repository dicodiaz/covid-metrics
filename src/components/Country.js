import PropTypes from 'prop-types';
import { Card, Col, Row } from 'react-bootstrap';
import { useNavigate } from 'react-router-dom';

const Country = ({ name, confirmedTotal, deathsTotal, recoveredTotal, index }) => {
  const navigate = useNavigate();

  return (
    <Card
      className={`rounded-0 h-100 cursor-pointer${
        index % 4 === 0 || (index - 3) % 4 === 0 ? ' bg-custom2' : ' bg-custom3'
      }`}
      onClick={() => navigate(`/details/${name}`)}
      data-testid="country"
    >
      <Card.Body className="d-flex flex-column justify-content-center">
        <Card.Title>{name}</Card.Title>
        <Row xs={1}>
          <Col>Confirmed: {confirmedTotal}</Col>
          <Col>Deaths: {deathsTotal}</Col>
          <Col>Recovered: {recoveredTotal}</Col>
        </Row>
      </Card.Body>
    </Card>
  );
};

Country.propTypes = {
  name: PropTypes.string.isRequired,
  confirmedTotal: PropTypes.number.isRequired,
  deathsTotal: PropTypes.number.isRequired,
  recoveredTotal: PropTypes.number.isRequired,
  index: PropTypes.number.isRequired,
};

export default Country;
