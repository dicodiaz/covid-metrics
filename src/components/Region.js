import PropTypes from 'prop-types';
import { Card, Col, Row } from 'react-bootstrap';

const Region = ({ name, confirmedTotal, deathsTotal, recoveredTotal }) => (
  <Card className="rounded-0 h-100 bg-region" data-testid="region">
    <Card.Body className="d-flex flex-column justify-content-center">
      <Row xs={2}>
        <Card.Title className="mb-0 d-flex flex-column justify-content-center">{name}</Card.Title>
        <Col>
          <Row xs={1}>
            <Col>Confirmed: {confirmedTotal}</Col>
            <Col>Deaths: {deathsTotal}</Col>
            <Col>Recovered: {recoveredTotal}</Col>
          </Row>
        </Col>
      </Row>
    </Card.Body>
  </Card>
);

Region.propTypes = {
  name: PropTypes.string.isRequired,
  confirmedTotal: PropTypes.number.isRequired,
  deathsTotal: PropTypes.number.isRequired,
  recoveredTotal: PropTypes.number.isRequired,
};

export default Region;
