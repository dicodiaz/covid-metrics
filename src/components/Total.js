import PropTypes from 'prop-types';
import { Col, Row } from 'react-bootstrap';

const Total = ({ covidDataTotal }) => {
  const {
    today_confirmed: confirmedTotal,
    today_deaths: deathsTotal,
    today_recovered: recoveredTotal,
    today_new_confirmed: confirmedToday,
    today_new_deaths: deathsToday,
    today_new_recovered: recoveredToday,
  } = covidDataTotal;
  return (
    <Row xs={1} className="mx-0 gx-0 gy-1">
      <Col className="bg-custom3">Total confirmed: {confirmedTotal}</Col>
      <Col>Total deaths: {deathsTotal}</Col>
      <Col className="bg-custom3">Total recovered: {recoveredTotal}</Col>
      <Col>Today&apos;s new confirmed: {confirmedToday}</Col>
      <Col className="bg-custom3">Today&apos;s new deaths: {deathsToday}</Col>
      <Col>Today&apos;s new recovered: {recoveredToday}</Col>
    </Row>
  );
};

Total.propTypes = {
  covidDataTotal: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
};

export default Total;
