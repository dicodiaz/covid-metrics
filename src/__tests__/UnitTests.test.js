import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Country from '../components/Country';
import Region from '../components/Region';
import SearchBar from '../components/SearchBar';
import Total from '../components/Total';

const [nameCountry, nameRegion, confirmedTotal, deathsTotal, recoveredTotal, index] = [
  'Colombia',
  'BogotÃ¡',
  100,
  1,
  99,
  0,
];
const covidDataTotalMock = {
  today_confirmed: 150,
  today_deaths: 1,
  today_recovered: 100,
  today_new_confirmed: 2,
  today_new_deaths: 0,
  today_new_recovered: 10,
};

describe('testing snapshots for my child components', () => {
  test('testing snapshot for Country.js', () => {
    const countryTree = render(
      <Router>
        <Country
          name={nameCountry}
          confirmedTotal={confirmedTotal}
          deathsTotal={deathsTotal}
          recoveredTotal={recoveredTotal}
          index={index}
        />
      </Router>,
    );
    expect(countryTree).toMatchSnapshot();
  });

  test('testing snapshot for Region.js', () => {
    const regionTree = render(
      <Region
        name={nameRegion}
        confirmedTotal={confirmedTotal}
        deathsTotal={deathsTotal}
        recoveredTotal={recoveredTotal}
        index={index}
      />,
    );
    expect(regionTree).toMatchSnapshot();
  });

  test('testing snapshot for Total.js', () => {
    const totalTree = render(<Total covidDataTotal={covidDataTotalMock} />);
    expect(totalTree).toMatchSnapshot();
  });

  test('testing snapshot for SearchBar.js', () => {
    const searchBarTree = render(<SearchBar value="" onChange={() => {}} clear={() => {}} />);
    expect(searchBarTree).toMatchSnapshot();
  });
});
