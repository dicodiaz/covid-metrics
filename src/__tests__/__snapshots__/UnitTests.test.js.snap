// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing snapshots for my child components testing snapshot for Country.js 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rounded-0 h-100 cursor-pointer bg-custom2 card"
        data-testid="country"
      >
        <div
          class="d-flex flex-column justify-content-center card-body"
        >
          <div
            class="card-title h5"
          >
            Colombia
          </div>
          <div
            class="row row-cols-1"
          >
            <div
              class="col"
            >
              Confirmed: 
              100
            </div>
            <div
              class="col"
            >
              Deaths: 
              1
            </div>
            <div
              class="col"
            >
              Recovered: 
              99
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rounded-0 h-100 cursor-pointer bg-custom2 card"
      data-testid="country"
    >
      <div
        class="d-flex flex-column justify-content-center card-body"
      >
        <div
          class="card-title h5"
        >
          Colombia
        </div>
        <div
          class="row row-cols-1"
        >
          <div
            class="col"
          >
            Confirmed: 
            100
          </div>
          <div
            class="col"
          >
            Deaths: 
            1
          </div>
          <div
            class="col"
          >
            Recovered: 
            99
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing snapshots for my child components testing snapshot for Region.js 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rounded-0 h-100 bg-custom2 card"
        data-testid="region"
      >
        <div
          class="d-flex flex-column justify-content-center card-body"
        >
          <div
            class="row row-cols-2"
          >
            <div
              class="mb-0 d-flex flex-column justify-content-center card-title h5"
            >
              Bogotá
            </div>
            <div
              class="col"
            >
              <div
                class="row row-cols-1"
              >
                <div
                  class="col"
                >
                  Confirmed: 
                  100
                </div>
                <div
                  class="col"
                >
                  Deaths: 
                  1
                </div>
                <div
                  class="col"
                >
                  Recovered: 
                  99
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rounded-0 h-100 bg-custom2 card"
      data-testid="region"
    >
      <div
        class="d-flex flex-column justify-content-center card-body"
      >
        <div
          class="row row-cols-2"
        >
          <div
            class="mb-0 d-flex flex-column justify-content-center card-title h5"
          >
            Bogotá
          </div>
          <div
            class="col"
          >
            <div
              class="row row-cols-1"
            >
              <div
                class="col"
              >
                Confirmed: 
                100
              </div>
              <div
                class="col"
              >
                Deaths: 
                1
              </div>
              <div
                class="col"
              >
                Recovered: 
                99
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing snapshots for my child components testing snapshot for SearchBar.js 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mx-0 gx-0 px-2 row"
      >
        <div
          class="col"
        >
          <input
            class="form-control"
            placeholder="Search..."
            value=""
          />
        </div>
        <div
          class="col-auto"
        >
          <button
            class="btn btn-danger"
            type="button"
          >
            Clear
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mx-0 gx-0 px-2 row"
    >
      <div
        class="col"
      >
        <input
          class="form-control"
          placeholder="Search..."
          value=""
        />
      </div>
      <div
        class="col-auto"
      >
        <button
          class="btn btn-danger"
          type="button"
        >
          Clear
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing snapshots for my child components testing snapshot for Total.js 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mx-0 gy-1 mb-2 row row-cols-1"
      >
        <div
          class="col"
        >
          Total confirmed: 
          150
        </div>
        <div
          class="bg-custom3 col"
        >
          Total deaths: 
          1
        </div>
        <div
          class="col"
        >
          Total recovered: 
          100
        </div>
        <div
          class="bg-custom3 col"
        >
          Today's new confirmed: 
          2
        </div>
        <div
          class="col"
        >
          Today's new deaths: 
          0
        </div>
        <div
          class="bg-custom3 col"
        >
          Today's new recovered: 
          10
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mx-0 gy-1 mb-2 row row-cols-1"
    >
      <div
        class="col"
      >
        Total confirmed: 
        150
      </div>
      <div
        class="bg-custom3 col"
      >
        Total deaths: 
        1
      </div>
      <div
        class="col"
      >
        Total recovered: 
        100
      </div>
      <div
        class="bg-custom3 col"
      >
        Today's new confirmed: 
        2
      </div>
      <div
        class="col"
      >
        Today's new deaths: 
        0
      </div>
      <div
        class="bg-custom3 col"
      >
        Today's new recovered: 
        10
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
